sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -414,8 +414,11 @@ def get_manifest_properties(self, repository, tag_or_digest, **kwargs):
         if _is_tag(tag_or_digest):
             tag_or_digest = self._get_digest_from_tag(repository, tag_or_digest)
 
+        manifest_properties = self._client.container_registry.get_manifest_properties(
+            repository, tag_or_digest, **kwargs
+        )
         return ArtifactManifestProperties._from_generated(  # pylint: disable=protected-access
-            self._client.container_registry.get_manifest_properties(repository, tag_or_digest, **kwargs),
+            manifest_properties.manifest, # type: ignore
```

Comment:
I'm surprised that this `# type: ignore` is necessary -- it looks like the type of `.manifest` should align with the type hint for `_from_generated`. Do you know if a Python 3-style type hint on `_from_generated` might help?
```python
def _from_generated(cls, generated: "ManifestAttributesBase", **kwargs) -> "ArtifactManifestProperties":
```